### PR TITLE
Datamodel

### PR DESCRIPTION
**Pull request by storres20**
# In this branch, I followed the next steps:

* Create a new app. Give it a meaningful name.
* Initialize your project with Git.
* Make sure that your project has Postgres database set up.
* Build your project schema.
  * Create and run the necessary migration files.
  * Table and column names should match the ERD diagram.
    * note: photo for users table should be a link to an image
  * Foreign keys should be included.
  * All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/migrate/20230224042902_add_user_ref_to_likes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
+  end
```
Comment (tresorsawasawa) [inline db/migrate/20230224042902_add_user_ref_to_likes.rb]: - [ ] According to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), kindly ensure that a `Like` belongs to an `:author` and not a `:user` so you should  refer the user table for the `foreign_key`. i.e.
```ruby
  def change
    add_reference :likes, :author, null: false, foreign_key: `{ to_table: 'users' }`
  end
```

File: db/migrate/20230224042343_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
```
Comment (tresorsawasawa) [inline db/migrate/20230224042343_add_user_ref_to_comments.rb]: - [ ] According to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), kindly ensure that a `Comment` belongs to an `:author` and not a `:user` so you should  refer the user table for the `foreign_key`. i.e.
```ruby
  def change
    add_reference :likes, :author, null: false, foreign_key: `{ to_table: 'users' }`
  end
```

File: db/migrate/20230224044021_add_user_ref_to_posts.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
+  end
```
Comment (tresorsawasawa) [inline db/migrate/20230224044021_add_user_ref_to_posts.rb]: - [ ] According to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), kindly ensure that a `Post` belongs to an `:author` and not a `:user` so you should  refer the user table for the `foreign_key`. i.e.
```ruby
  def change
    add_reference :likes, :author, null: false, foreign_key: `{ to_table: 'users' }`
  end
```

- [ ] After you have implemented all the above changes, kindly ensure that you update your migrations as they will create `author_id` and a corresponding index. You can refer this **[article](https://stackoverflow.com/questions/1992019/how-can-i-rename-a-database-column-in-a-ruby-on-rails-migration)**  to help you how to rename a column

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @storres20 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @storres20, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**